apps/dashboard/pkg/migration/conversion/v1beta1_to_v2alpha1.go

```diff
@@ -1709,6 +1717,10 @@ func buildGroupByVariable(ctx context.Context, varMap map[string]interface{}, co
 
 		// Resolve Grafana datasource UID when type is "datasource" and UID is empty
 		datasourceUID = resolveGrafanaDatasourceUID(datasourceType, datasourceUID)
+	} else if dsStr, ok := datasource.(string); ok {
```

Comment:
I see this code in 3 places. Should we move it to a function all the DS resolution?